src/data.rs

```diff
@@ -105,6 +110,37 @@ mod array_impls {
         })))
     }
 
+    pub(crate) fn create_boolean_array_with_errors<DT: CellType + DataType + Debug>(
+        data: &Range<DT>,
+        col: usize,
+        offset: usize,
+        limit: usize,
+    ) -> (Arc<dyn Array>, CellErrors) {
+        let mut cell_errors = CellErrors { errors: vec![] };
```

Comment:
(just style, I'm fine with the current form as well) We don't need the `CellErrors` struct here:

```suggestion
        let mut cell_errors = vec![];
```

Reply:
Updated